crates/utils/src/tracing/span_ext.rs

```diff
@@ -20,29 +21,53 @@ impl ToValue for Digest {
     }
 }
 
-impl ToValue for f64 {
-    fn to_value(&self) -> Value {
-        (*self).into()
-    }
-}
-
 impl ToValue for BlockNumber {
     fn to_value(&self) -> Value {
         i64::from(self.as_u32()).into()
     }
 }
 
-impl ToValue for u32 {
-    fn to_value(&self) -> Value {
-        i64::from(*self).into()
-    }
+/// Generates `impl ToValue` blocks for types that are `ToString`.
+macro_rules! impl_to_string_to_value {
+    ($($t:ty),*) => {
+        $(
+            impl ToValue for $t {
+                fn to_value(&self) -> Value {
+                    self.to_string().into()
+                }
+            }
+        )*
+    };
 }
+impl_to_string_to_value!(IpAddr, &str);
 
-impl ToValue for i64 {
-    fn to_value(&self) -> Value {
-        (*self).into()
-    }
+/// Generates `impl ToValue` blocks for integer types.
+macro_rules! impl_int_to_value {
+    ($($t:ty),*) => {
+        $(
+            impl ToValue for $t {
+                fn to_value(&self) -> Value {
+                    i64::from(*self).into()
+                }
+            }
+        )*
+    };
+}
+impl_int_to_value!(u16, u32, i64);
+
+/// Generates `impl ToValue` blocks for types that are `Into<Value>`.
+macro_rules! impl_to_value {
+    ($($t:ty),*) => {
+        $(
+            impl ToValue for $t {
+                fn to_value(&self) -> Value {
+                    (*self).into()
+                }
+            }
+        )*
+    };
```

Comment:
Might be more idiomatic as a blanket implementation:

```rust
impl<T> ToValue for T 
where T: Into<Value> + Clone {
    fn to_value(&self) -> Value {
        self.clone().into()
    }
}
```

The downside is this blanket impl doesn't let you choose specific types.

Reply:
I think at that point you lose the benefit of having the separate `ToValue` trait which is to be able to pass in types without first converting them to `f64` or `i64` etc.

Reply:
Yeah it does encourage poor practices. Let's leave it as is 👍